### PR TITLE
use trusty distribution

### PR DESCRIPTION
**Pull request by wangjoshuah**
Upgrade travis-ci `travis.yml` to use `trusty` distribution since that is their new standard.
The gradle version of their linux distribution is gradle `4.0.1` so I upgraded our gradle wrapper to that.
That required we upgrade our nebula dependency to `3.2.0`.
Drop support for JDK 7 until travis-ci can fix that.

### CODE REVIEW

File: core-api/build.gradle [hunk 1]
```diff
@@ -2,14 +2,14 @@ dependencies {
...
-    provided group: 'com.google.code.findbugs', name: 'jsr305', version: findbugsVersion
+    compile group: 'com.google.code.findbugs', name: 'annotations', version: findbugsVersion
+    compile group: 'com.google.code.findbugs', name: 'jsr305', version: findbugsVersion
 
     // an assortment of json parsers
-    provided group: 'com.google.code.gson', name: 'gson', version: gsonVersion, optional
-    provided group: 'org.json', name: 'json', version: jsonVersion, optional
-    provided group: 'com.googlecode.json-simple', name: 'json-simple', version: jsonSimpleVersion, optional
-    provided group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: jacksonVersion, optional
+    compile group: 'com.google.code.gson', name: 'gson', version: gsonVersion, optional
```
Comment (vraja2) [inline core-api/build.gradle]: Changing these to `compile` will result in all of these JSON libraries being packaged with the SDK which will increase bloat. `provided` was used to give that dependency flexibility to the consumer.

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,14 @@
 language: java
-dist: precise
+dist: trusty
+sudo: required
```
Comment (vraja2) [inline .travis.yml]: any reason we need root privileges?

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
 jdk:
-  - openjdk7
```
Comment (vraja2) [inline .travis.yml]: I think `openjdk7` should still be supported.

**Comment by wangjoshuah** [comment]
@vraja2 travis-ci.org is ending support for `dist: precise`. We are migrating to `dist: trusty` do ensure support. https://blog.travis-ci.com/2017-08-31-trusty-as-default-status

The `trusty` distribution uses `gradle 4.0.1` which is incompatible with our current version of `nebula`. We will use `nebula 3.2.0` to allow us to continue testing all parsers. https://github.com/nebula-plugins/gradle-extra-configurations-plugin

**Pull request merged**